jsonforms-editor/src/properties/renderers/RuleEditorRenderer.tsx

```diff
@@ -66,35 +111,48 @@ const RuleEditor: React.FC<ControlProps> = (props) => {
       setInvalidJson(true);
     }
   }, [handleChange, model, path]);
+  console.log(data);
 
   const isValid = errors.length === 0 && !invalidJson;
+  const classes = useStyles();
   return (
-    <div>
-      <Typography>Rule</Typography>
-      <MonacoEditor
-        language='json'
-        editorWillMount={configureEditor}
-        editorDidMount={setModel}
-        height={200}
-        options={{
-          formatOnPaste: true,
-          formatOnType: true,
-          automaticLayout: true,
-        }}
-      />
-      <Grid container direction='row' spacing={2} alignItems='center'>
-        <Grid item>
-          <Button variant='contained' onClick={onSubmitRule}>
-            Apply
-          </Button>
-        </Grid>
-        <Grid item>
-          <FormHelperText error={true} hidden={isValid}>
-            {errors.length !== 0 ? errors : invalidJsonMessage}
-          </FormHelperText>
-        </Grid>
-      </Grid>
-    </div>
+    <Accordion defaultExpanded={!!data}>
+      <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+        <Typography>Rule</Typography>
+      </AccordionSummary>
+      <AccordionDetails>
+        <div className={classes.editorRoot}>
+          <FormHelperText error={false}>{ruleDescription}</FormHelperText>
+          <ShowMoreLess className={classes.showMore}>
```

Comment:
why do we want to pass the styling down and not define it in the component?

Reply:
The style only contains a padding and I don't really see it as component-specific. It's a style of the container. I added the class to the ShowMoreLess component to avoid adding a div around it.